src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,187 @@
+"""GitHub App Installation Access Token Credential Plugin.
+
+This module defines a credential plugin for making use of the
+GitHub Apps mechanism, allowing authentication via GitHub App
+installation-scoped access tokens.
+
+Functions:
+
+- :func:`extract_github_app_install_token`: Generates a GitHub App
+  Installation token.
+- ``github_app_lookup``: Defines the credential plugin interface.
+"""
+
+from typing import NotRequired, TypedDict, Unpack
+
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+from github import Auth as Auth, Github
+
+from .plugin import CredentialPlugin
+
+
+__all__ = ('github_app_lookup',)  # noqa: WPS410
+
+
+class FieldDict(TypedDict):
+    """A single UI field schema."""
+
+    id: str
+    label: str
+    type: str
+    format: NotRequired[str]
+    secret: NotRequired[bool]
+    multiline: NotRequired[bool]
+    help_text: str
+    default: NotRequired[str]
+
+
+class MetadataDict(TypedDict):
+    """Schema for input metadata."""
+
+    id: str
+    label: NotRequired[str]
+    type: str
+    help_text: str
+
+
+class GitHubAppInputs(TypedDict):
+    """Schema for a collection of plugin input fields."""
+
+    fields: list[FieldDict]
+    metadata: list[MetadataDict]
+    required: list[str]
+
+
+github_app_inputs: GitHubAppInputs = {
+    'fields': [
+        {
+            'id': 'github_api_url',
+            'label': _('GitHub API endpoint URL'),
+            'type': 'string',
+            'help_text': _(
+                'Specify the GitHub API URL here. In the case of an Enterprise: '
+                ' https://github.acme.com/api/v3 (self-hosted) '
+                'or https://github.acme.com/api/v3 (cloud)',
```

Comment:
This offers examples with parts of URLs that are obviously placeholders:
```suggestion
                ' https://gh.your.org/api/v3 (self-hosted) '
                'or https://api.SUBDOMAIN.ghe.com (cloud)',
```
These correspond to the placeholder GitHub uses in their own docs too:
* https://docs.github.com/en/enterprise-cloud@latest/rest/meta/meta?apiVersion=2022-11-28#github-api-root--code-samples
* https://docs.github.com/en/enterprise-server@3.15/rest/meta/meta?apiVersion=2022-11-28#get-all-api-versions